src/data/fintype/basic.lean

```diff
@@ -1636,3 +1636,12 @@ variables [fintype α] [decidable_eq α]
 count_eq_one_of_mem finset.univ.nodup (finset.mem_univ _)
 
 end multiset
+
+namespace fintype
+variables [fintype α] [complete_lattice β]
+
+lemma sup_eq_supr (f : α → β) : finset.univ.sup f = supr f :=
```

Comment:
```suggestion
/-- A special case of `finset.sup_eq_supr` with a better RHS. -/
lemma sup_eq_supr (f : α → β) : finset.univ.sup f = supr f :=
```
I'd also be tempted to rename this to `finset.sup_univ_eq_supr` to match the other lemma.

Reply:
I've PR'd this as #8397, along with the `inf` version.